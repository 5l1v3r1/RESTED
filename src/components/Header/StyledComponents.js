import styled from 'styled-components';

/* eslint-disable import/prefer-default-export */
export const StyledHeader = styled.header`
  margin-top: 20px;
  margin-bottom: 30px;

  img {
    margin-top: -7px;
    margin-right: 10px;
  }
`;

