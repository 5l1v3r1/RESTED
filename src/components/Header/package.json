{
  "name": "Header",
  "version": "0.0.0",
  "main": "./Header.js"
}

