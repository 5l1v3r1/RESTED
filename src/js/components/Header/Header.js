import React from 'react';

export default function Header() {
  return (
    <header>
      <h1>
        <img
          className="logo"
          alt="RESTED logo"
          height="40"
          src="img/rested-logo.png"
        />
        <span>RESTED</span>
      </h1>
    </header>
  );
}

