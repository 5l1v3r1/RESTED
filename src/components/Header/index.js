import React from 'react';
import { StyledHeader } from './StyledComponents';

export default function Header() {
  return (
    <StyledHeader>
      <h1>
        <img
          className="logo"
          role="presentation"
          height="40"
          src="img/rested-logo.png"
        />
        <span>RESTED</span>
      </h1>
    </StyledHeader>
  );
}

