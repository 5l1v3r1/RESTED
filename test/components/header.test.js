import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';

/* eslint-disable import/no-unresolved */
import Header from 'components/Header';

jest.mock('react-dom');

test('Header renders correctly', () => {
  const tree = renderer.create(
    <Header />,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

test('Header contains the RESTED logo', () => {
  const tree = mount(
    <Header />,
  );
  expect(tree.find('img').prop('src')).toBe('img/rested-logo.png');
});

test('Header contains the RESTED name', () => {
  const tree = mount(
    <Header />,
  );
  expect(tree.find('h1 span').prop('children')).toBe('RESTED');
});

