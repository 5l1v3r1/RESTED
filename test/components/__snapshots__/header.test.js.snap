exports[`test Header renders correctly 1`] = `
<header>
  <h1>
    <img
      alt="RESTED logo"
      className="logo"
      height="40"
      src="img/rested-logo.png" />
    <span>
      RESTED
    </span>
  </h1>
</header>
`;
